### PR TITLE
Normalization and exit frames

### PR DESCRIPTION
**Pull request by mbrysa**

### CODE REVIEW

**Comment by mbrysa** [comment]
The current way of detecting and normalizing exit frames requires the frame's `source_file` to be valid, i.e. it won't work when FAF passes `"??"` when it couldn't retrace the frame. Do you think this is what we want? If we removed the `source_file` requirement or added  `"??"`, there could be false positives.

File: include/core/thread.h [hunk 1]
```diff
@@ -123,6 +123,9 @@ struct sr_core_thread *
 sr_core_thread_append(struct sr_core_thread *dest,
                       struct sr_core_thread *item);
 
+bool
+sr_core_is_exit_frame(struct sr_core_frame *frame);
```
Comment (mmilata) [inline include/core/thread.h]: Please rename to `sr_core_thread_is_exit_frame` to be consistent with the rest of the function names.

**Comment by mmilata** [comment]
Otherwise looks fine, I'll merge it after the rename.

**Comment by mbrysa** [comment]
Renamed.
What is your opinion on adding `"??"` to possible `source_file`s in `sr_core_thread_is_exit_frame` and `sr_gdb_is_exit_frame`?

**Comment by mmilata** [comment]
Merged, thanks!

Regarding your question I would be in favor of some more systematic solution - either ignoring the `source_file` altogether, or introducing a new function/flag that would ignore the source file.

**Pull request merged**